jfrog-cli/artifactory/utils/nuget/solution/solution_test.go

```diff
@@ -24,3 +30,128 @@ func TestEmptySolution(t *testing.T) {
 		t.Errorf("Expecting: \n%s \nGot: \n%s", expectedString, buildInfoString)
 	}
 }
+
+func TestParseSln(t *testing.T) {
+	regExp, err := utils.GetRegExp(`Project\("(.*)\nEndProject`)
+	if err != nil {
+		t.Error(err)
+	}
+	pwd, err := os.Getwd()
+	if err != nil {
+		t.Error(err)
+	}
+
+	testsDataDir := filepath.Join(pwd, "testsdata")
+
+	tests := []struct {
+		name     string
+		slnPath  string
+		expected []string
+	}{
+		{"oneproject", filepath.Join(testsDataDir, "oneproject.sln"), []string{`Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "packagesconfig", "packagesconfig.csproj", "{D1FFA0DC-0ACC-4108-ADC1-2A71122C09AF}"
+EndProject`}},
+		{"multiProjects", filepath.Join(testsDataDir, "multiprojects.sln"), []string{`Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "packagesconfigmulti", "packagesconfig.csproj", "{D1FFA0DC-0ACC-4108-ADC1-2A71122C09AF}"
+EndProject`, `Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "packagesconfiganothermulti", "test\packagesconfig.csproj", "{D1FFA0DC-0ACC-4108-ADC1-2A71122C09AF}"
+EndProject`}},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			results, err := parseSlnFile(test.slnPath, regExp)
+			if err != nil {
+				t.Error(err)
+			}
+
+			replaceCarriageSign(results)
+
+			if !reflect.DeepEqual(test.expected, results) {
+				t.Error(fmt.Sprintf("Expected %s, got %s", test.expected, results))
+			}
+		})
+	}
+}
+
+func TestParseProject(t *testing.T) {
```

Comment:
I think it it best to use resource files instead of having the sln content as part of the test code.
The code will be easier to read and maintain in my opinion if the sln content is in a testsdata resource.
The same goes for the *TestGetProjectsFromSlns* test.

Reply:
Per our discussion, resolving this